blog/vscode-remote-development-setup-and-usage.mdx

```diff
@@ -0,0 +1,263 @@
+
+During specific use cases, I sometime encounter issues with port forwarding that
+force me to get back to my laptop’s keyboard, for example when using
+[Docker](https://www.docker.com/) with [Traefik](https://traefik.io/traefik/),
+VS Code is sometime confused about what port to forward and where to forward it
+to.
+
+On the hardware side, this method renders your laptop (or host) a bit less
+useful, as most of the time you’ll work at home you won’t touch it much, besides
+turning it on or off. We will see in the next part other methods I tried that
+actually uses both machines actively.
+
+On the management side, depending on how you manage your local machine, and how
+(if) you use it outside of work, you’ll have to take extra care of not mixing
+work and personal things, which can be extra tricky for some, especially if
+you’re pretty new to working at home.
+
+## Comparison with other methods I have tried
+
+### Why not use a dock ?
+
+Most docks are really great at what they are meant to do. Unfortunately, this
+doesn’t include sharing peripherals between devices, which was my goal here. I
+still use my dock, however in my current setup, since I use the same keyboard /
+mouse for both my machines, it would mean plugging / unplugging those everyday
+which is kind of painful.
+
+### Using both machine efficiently: Synergy / Barrier
+
+This was from far my most preferred method.
+[Synergy](https://symless.com/synergy) (and its open source clone
+[Barrier](https://github.com/debauchee/barrier)) is a keyboard and mouse sharing
+software, that enable a single pair of peripherals to work on many different
+machines at the same time. Unlike VS Code remote (or SSH) only, using Synergy
+you are able to actually use every device’s UI as well, and coupled to VS Code
+Remote, you can basically use “any” number of machines the same way you’ll be
+using one.
+
+Unfortunately, I had a bug on Ubuntu using Barrier that made me stop using this
+method some times ago as I couldn’t find a solution. If this issue was to be
+fixed however, I would hop in the Barrier train any day !
+
+### Another point of view: the Virtual Machine way, and Tmux
+
+That’s a solution that requires a bit more setup, and isn’t really compatible
+with VS Code either, but might be interesting for Vi / Vim and terminal power
+users out there. Tmux is a terminal multiplexer, a software capable of running
+multiple terminal session in a single windows. This also allows terminal session
+to be “saved” and accessed from another device, meaning using terminal based
+editors, your could pickup your work directly right where you left it
+seamlessly, from pretty much any terminal on OSs supporting tmux (even from your
+phone!). Coupling Tmux with a Virtual-machine hosted source code, it would be
+one of the best method I can think of in terms of accessibility and portability
+(given you always have an internet connection where you go, which is not an hard
+requirement to satisfy in 2023).
```

Comment:
```suggestion
That’s a solution that requires a bit more setup, and isn’t really compatible
with VS Code either, but might be interesting for Vi / Vim and terminal power
users out there. Tmux is a terminal multiplexer, a software capable of running
multiple terminal session in a single windows. This also allows terminal session
to be “saved” and accessed from another device, meaning using terminal based
editors, your could pickup your work right where you left it seamlessly, from 
pretty much any terminal on OSs supporting tmux (even from your phone!). 
Coupling Tmux with a Virtual-machine hosted source code, it would be
one of the best method I can think of in terms of accessibility and portability
(given you always have an internet connection where you go, which is not a hard
requirement to satisfy in 2023).
```